app/Http/Controllers/ClassController.php

```diff
@@ -67,11 +75,16 @@ public function update(Request $request, string $id)
     {
         $data = $request->validate([
             'class_name' => 'required|alpha_num:ascii|max:255',
-            'capacity' => 'required|numeric|integer|min:1', 
+            'capacity' => 'required|numeric|integer|min:1',
             'price' => 'required|decimal:0,2',
             'time_From' => 'required|date|after:tomorrow',
             'time_to' => 'required|date|after:time_From',
+            'image' => 'mimes:png,jpg,jpeg|max:2048',
         ]);
+      
+        $file_name = ($request['image'] == null) ? $request['old_image']:$this->uploadFile($request['image'], 'assets/images');
```

Comment:
although it is not wrong, but you can benefit from if you didn't send a specific field, it didn't change

what i mean by this 
for example 

if  you send the following data to `update()` method of the model
```php
$data = [
    'class_name' => 'test',
    'capacity' => 10,
    'price' => 13,
];
```

the `time_Form`, `time_to` & `image` won't update